include/fastdds/rtps/security/logging/Logging.h

```diff
@@ -0,0 +1,115 @@
+// Copyright 2020 Canonical ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*!
+ * @file Logging.h
+ */
+#ifndef _FASTDDS_RTPS_SECURITY_LOGGING_LOGGING_H_
+#define _FASTDDS_RTPS_SECURITY_LOGGING_LOGGING_H_
+
+#include "fastdds/rtps/security/logging/LogOptions.h"
+#include "fastdds/rtps/security/exceptions/SecurityException.h"
+
+namespace eprosima {
+namespace fastrtps {
+namespace rtps {
+namespace security {
+
+/**
+ * @brief The LoggerListener class
+ */
+class LoggerListener
+{
+  LoggerListener() = default;
+  ~LoggerListener() = default;
+};
+
+/**
+ * @brief Base class for all security logging plugins.
+ */
+class Logging
+{
+public:
+
+  Logging();
+  virtual ~Logging() = default;
+
+  /**
+   * @brief set_log_options
+   * @param log_options
+   * @return TRUE if successful
+   */
+  bool set_log_options(const LogOptions& log_options, SecurityException& exception);
```

Comment:
My reading of table 40 (section 8.6.2.2 of the DDS-Security spec) is that the exception is a possible output. i.e. it's possible for this function to throw a SecurityException, not requiring that it be passed in. Do you disagree?

Reply:
The specification explicitly refers to exceptions as 'parameter' and this is what's done in [other places of the code base](https://github.com/ubuntu-robotics/Fast-RTPS/blob/master/src/cpp/fastrtps_deprecated/security/accesscontrol/Permissions.h#L40).

Reply:
Ah that does indeed clarify it. Weird, but established. Thanks!